devolutions-gateway/src/api/update.rs

```diff
@@ -0,0 +1,63 @@
+use axum::extract::Query;
+use axum::Json;
+use hyper::StatusCode;
+
+use devolutions_agent_shared::{get_updater_file_path, ProductUpdateInfo, UpdateJson, VersionSpecification};
+
+use crate::extract::UpdateScope;
+use crate::http::{HttpError, HttpErrorBuilder};
+
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct UpdateQueryParam {
+    version: VersionSpecification,
+}
+
+#[cfg_attr(feature = "openapi", derive(utoipa::ToSchema))]
+#[derive(Serialize)]
+pub(crate) struct UpdateResponse {}
+
+/// Starts Devolutions Gateway update process
+#[cfg_attr(feature = "openapi", utoipa::path(
+    get,
+    operation_id = "Update",
+    tag = "Update",
+    path = "/jet/update",
+    responses(
+        (status = 200, description = "Update request has been processed successfully", body = UpdateResponse),
+        (status = 400, description = "Bad request"),
+        (status = 401, description = "Invalid or missing authorization token"),
+        (status = 403, description = "Insufficient permissions"),
+        (status = 500, description = "Agent updater service is malfunctioning"),
+        (status = 503, description = "Agent updater service is unavailable"),
+    ),
+    security(("scope_token" = ["gateway.update"])),
+))]
+pub(super) async fn start_update(
```

Comment:
note: The naming suggests that the update procedure will actually be started by the function, but in fact all we do is updating the `update.json`. I think this could be clarified in the name. It would also be nice to document the logic a little bit.